setup-cabal.sh

```diff
@@ -14,6 +14,23 @@ sed -e 's/inline-c ==.*,/inline-c ==0.9.0.0,/g' \
     > cabal.project.freeze
 fi
 
+case "$(uname)" in
+  "Darwin")
+      TOTAL_MEM_GB=`sysctl hw.physmem | awk '{print int($2/1024/1024/1024)}'`
+      NUM_CPU=$(sysctl -n hw.ncpu)
+    ;;
+  "Linux")
+      TOTAL_MEM_GB=`grep MemTotal /proc/meminfo | awk '{print int($2/1024/1024)}'`
+      NUM_CPU=$(nproc --all)
+    ;;
+esac
+
+USED_MEM_GB=`echo $TOTAL_MEM_GB | awk '{print int(($1 + 1) / 2)}'`
+USED_NUM_CPU=`echo $NUM_CPU | awk '{print int(($1 + 1) / 2)}'`
+USED_NUM_CPU=`echo $USED_MEM_GB $USED_NUM_CPU | awk '{if($1<x$2) {print $1} else {print $2}}'`
+USED_MEM_GB=`echo $USED_NUM_CPU | awk '{print ($1)"G"}'`
+USED_MEMX2_GB=`echo $USED_NUM_CPU | awk '{print ($1 * 2)"G"}'`
```

Comment:
Hasktorch-package seems to need 1.5GB of memory per core. (This line sets 2GB memory per core for margin.)